examples/mnist/train.cpp

```diff
@@ -79,10 +79,12 @@ static void train_lenet(const std::string& data_dir_path) {
 
   progress_display disp(static_cast<unsigned long>(train_images.size()));
   timer t;
-  int minibatch_size = 10;
+  int minibatch_size = 16;
   int num_epochs     = 30;
 
-  optimizer.alpha *= static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size));
+  optimizer.alpha *=
+    std::min(tiny_dnn::float_t(4),
+             static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size)));
```

Comment:
I'm not sure how this change is needed. Aside from that, `std::sqrt` function has some overloads.
http://en.cppreference.com/w/cpp/numeric/math/sqrt, so rather than casting returned value, casting int argument to `tiny_dnn::float_t` might be another way to suppress a warning.